src/function/gds/all_shortest_paths.cpp

```diff
@@ -140,12 +145,10 @@ class AllSPDestinationsEdgeCompute : public EdgeCompute {
         PathMultiplicities* multiplicities)
         : frontierPair{frontierPair}, multiplicities{multiplicities} {};
 
-    void edgeCompute(nodeID_t boundNodeID, std::span<const nodeID_t> nbrIDs,
-        std::span<const relID_t>, SelectionVector& mask, bool) override {
-        size_t activeCount = 0;
-        mask.forEach([&](auto i) {
+    void edgeCompute(nodeID_t boundNodeID, GraphScanState::Chunk& resultChunk, bool) override {
+        resultChunk.filter([&](auto nbrNodeID, auto /*edgeID*/) {
```

Comment:
I am trying to understand what this filter does. Looking at graph.h, it seems to update the selection vector of the internal scanstate but why? Shouldn't this just be a blind forEach? When do we need to filter the selection vector?

Reply:
It's because we're re-using the SelectionVector in the chunk to set the next frontier. The filter function acts like forEach, but also updates the SelectionVector to include only the neighbours for which the function passed to filter returns true.
Maybe filter isn't the ideal name, but I thought that this was better than manually modifying the selection vector.

https://github.com/kuzudb/kuzu/blob/63840d0593bbd894e7c27258a289bff8c0844252/src/function/gds/gds_task.cpp#L15-L17
(the above is slightly different in this PR, but github doesn't show the inline code chunk for PR links)

Reply:
I've updated it to have edgeCompute return an `std::vector<nodeID_t>` instead for all the active nodes. The cost of this seems to be quite small compared to the rest of the work being done (benchmarking some of the algorithms currently implemented I couldn't find a measurable difference).